lib/instrumentation/grpc-js/grpc.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const EXTERNAL = require('../../metrics/names').EXTERNAL
+
+module.exports = function instrument(shim) {
+  const callStream = shim.require('./build/src/call-stream')
+  shim.wrap(callStream.Http2CallStream.prototype, 'start', (shim, original) => {
+    return function wrappedStart() {
+      const activeSegment = shim.getActiveSegment()
+      if (!activeSegment) {
+        return original.apply(this, arguments)
+      }
+
+      const channel = this.channel
+      const authorityName = channel.defaultAuthority
+
+      const segment = shim.createSegment({
+        name: `External/${authorityName}${this.methodName}`,
+        opaque: true,
+        recorder: recordExternal(authorityName, this.methodName)
+      })
+
+      return shim.applySegment(callStart, segment, true, this, arguments)
+
+      function callStart() {
+        const args = shim.argsToArray.apply(shim, arguments)
+
+        const transaction = segment.transaction
+
+        const originalMetadata = args[0]
+        const nrMetadata = originalMetadata.clone()
+
+        const outboundAgentHeaders = Object.create(null)
+        if (shim.agent.config.distributed_tracing.enabled) {
+          transaction.insertDistributedTraceHeaders(outboundAgentHeaders)
+        } else {
+          shim.logger.debug('Distributed tracing disabled by instrumentation.')
+        }
+
+        Object.keys(outboundAgentHeaders).forEach((key) => {
```

Comment:
nit but i think this is only applicable when DT is enabled as it creates new headers right?

Reply:
Ah, oops, you're right, will do.